api/server/http-rest/server.go

```diff
@@ -0,0 +1,112 @@
+package http_rest
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/api/server/middleware"
+	"github.com/prysmaticlabs/prysm/v5/runtime"
+)
+
+var _ runtime.Service = (*Server)(nil)
+
+// restHandler is a functional interface that implements the rest handler functionality.
+type restHandler func(
+	h http.HandlerFunc,
+	w http.ResponseWriter,
+	req *http.Request,
+)
+
+// Config parameters for setting up the http-rest service.
+type config struct {
+	httpAddr       string
+	allowedOrigins []string
+	muxHandler     restHandler
+	router         *mux.Router
+	timeout        time.Duration
+}
+
+// Server serves HTTP JSON traffic.
+type Server struct {
+	cfg          *config
+	server       *http.Server
+	cancel       context.CancelFunc
+	ctx          context.Context
+	startFailure error
+}
+
+// New returns a new instance of the Server.
+func New(ctx context.Context, opts ...Option) (*Server, error) {
+	g := &Server{
+		ctx: ctx,
+		cfg: &config{},
+	}
+	for _, opt := range opts {
+		if err := opt(g); err != nil {
+			return nil, err
+		}
+	}
+
+	if g.cfg.router == nil {
+		return nil, errors.New("router option not configured")
+	}
+
+	corsMux := middleware.CorsHandler(g.cfg.allowedOrigins).Middleware(g.cfg.router)
+	// TODO: actually use the timeout config provided
+	g.server = &http.Server{
+		Addr:              g.cfg.httpAddr,
+		Handler:           corsMux,
+		ReadHeaderTimeout: time.Second,
+	}
+	if g.cfg.muxHandler != nil { // rest APIS and Web UI registration
```

Comment:
This comment is weirdly placed

Reply:
you know what this is not even needed lol good reminder